realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -1719,4 +1720,29 @@ public void testFinalizerThread() throws NoSuchFieldException, IllegalAccessExce
                     + (totalNumberOfReferences - references.size()) + " out of " + totalNumberOfReferences);
         }
     }
+
+    // Test close Realm in another thread different from where it is created.
+    public void testCloseRealmInDifferentThread() throws InterruptedException {
+        final AtomicBoolean threadReady = new AtomicBoolean(false);
+
+        final Thread thatThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    testRealm.close();
+                    fail("Close realm in a different thread should throw IllegalStateException.");
```

Comment:
This fail will not report in JUnit, you need to capture it & throw it within the testcase's Thread 
ex: https://github.com/realm/realm-java/blob/nh-async-query/realm/src/androidTest/java/io/realm/RealmAsyncQueryTests.java#L241